internal/services/gateway/action/project.go

```diff
@@ -584,3 +592,62 @@ func (h *ActionHandler) getRemoteRepoAccessData(ctx context.Context, linkedAccou
 
 	return user, rs, la, nil
 }
+
+func (h *ActionHandler) GetRepoInfo(ctx context.Context, repoPath string, remoteSourceRef string) (*gitsource.RepoInfo, error) {
```

Comment:
This function could also be reused here: https://github.com/agola-io/agola/pull/303/files#diff-221e5cf1e1a163f665e4da8d541667d3a1c5179ee2923ab6824782447867c090L107-L115 and other places.

You should not get the curUserID inside it but pass it as parameter.

Or you should just don't create it and just add the code directly in `RefreshRemoteRepositoryInfo`